src/exec/execute_pipeline.c

```diff
@@ -0,0 +1,49 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   execute_pipeline.c                                 :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: reasuke <reasuke@student.42tokyo.jp>       +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/09/13 16:15:45 by reasuke           #+#    #+#             */
+/*   Updated: 2024/09/13 17:34:55 by reasuke          ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <errno.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "ast.h"
+#include "exec_internal.h"
+#include "utils.h"
+
+#define PIPE_IN 0
+#define PIPE_OUT 1
+
```

Comment:
this is merely a suggestion

```suggestion
```suggestion
 pipeline         : command
                  | command'|' pipeline
```
```